proof/crefine/RISCV64/Detype_C.thy

```diff
@@ -1668,16 +1650,17 @@ proof -
     by (clarsimp simp: rf_sr_def cstate_relation_def Let_def)
   hence "cpspace_relation ?ks' (underlying_memory (ksMachineState s)) ?th_s"
     unfolding cpspace_relation_def
-    using cendpoint_relation_restrict [OF D.valid_untyped invs rl]
-      cnotification_relation_restrict [OF D.valid_untyped invs rl]
+    using cendpoint_relation_restrict [OF D.valid_untyped invs sym_refs rl]
+      cnotification_relation_restrict [OF D.valid_untyped invs sym_refs rl]
     using  cmap_array[simplified bit_simps]
     apply -
     apply (elim conjE)
     apply ((subst lift_t_typ_region_bytes,
-            rule cm_disj cm_disj_tcb cm_disj_cte cm_disj_user cm_disj_device
-            , assumption +,
-            simp_all add: objBits_simps' bit_simps
-                          heap_to_user_data_restrict heap_to_device_data_restrict)[1])+ \<comment> \<open>waiting ...\<close>
+           rule cm_disj cm_disj_tcb cm_disj_cte cm_disj_user cm_disj_device, assumption+,
+           simp_all add: objBits_simps' bit_simps scBits_simps
+                         heap_to_user_data_restrict heap_to_device_data_restrict)[1],
+           (rule_tac y="2^7" in order_trans, simp,
+            fastforce intro!: power_increasing_iff[THEN iffD2] scBits_at_least_7)?)+ \<comment> \<open>waiting ...\<close>
```

Comment:
Wow, it can't have been fun getting this working!
It would be very nice to split this up into smaller, more understandable steps. However, considering that it was pretty bad already I wouldn't suggest putting too much effort into it.
However, could we make the comment a bit more specific? I'm assuming it means that this is a long running step, but I always find it more useful if comments like that provide a rough idea of how long the command is expected to take.

Reply:
I've seen some comments that say "takes long" or "60 seconds" take very little time, so I think some of the older comments are a bit out of whack. Sometimes I just remove them, but if they're vague, it's hard to know what to change them to if it might improve soon. But here I think it's only about 10 seconds, so I could put "10 seconds or so"